File: backend_springboot_kotlin/src/main/kotlin/com/davitm/librarymanager/models/Book.kt
package com.davitm.librarymanager.models
import jakarta.persistence.*

@Entity
data class Book(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    var title: String = "",
    var author: String = "",
    var publishYear: Int = 0,
    var available: Boolean = true
)

File: frontend_flutter/lib/domain/entities/book.dart
import 'package:equatable/equatable.dart';

class Book extends Equatable {
  final int? id;
  final String title;
  final String author;
  final int publishYear;
  final bool available;

  const Book({
    this.id,
    required this.title,
    required this.author,
    required this.publishYear,
    required this.available,
  });

  @override
  List<Object?> get props => [id, title, author, publishYear, available];
}
